src/GameObjectCounter.h

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+class GameObjectCounter {
+public:
+    static GameObjectCounter* get_instance();
+    int assign_id() const;
+    int id() const;
```

Comment:
This class should probably have some comments. Explaning it's purpose and the difference between `assign_id()` and `id()`